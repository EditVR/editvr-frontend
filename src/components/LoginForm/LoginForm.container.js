/**
 * @file LoginForm.container.js
 * Exports a redux-connected LoginForm component.
 */

import { connect } from 'react-redux';

import LoginForm from './LoginForm';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = ({ user }) => ({
  user
});

export default connect(mapState, mapDispatchToProps)(LoginForm);
