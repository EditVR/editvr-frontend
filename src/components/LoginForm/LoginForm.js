/**
 * @file LoginForm.js
 * Exports a component that allows users to log into EditVR.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextField, Button, withStyles } from '@material-ui/core';

import { FormLoading, FormMessage } from '../';
import LoginFormStyles from './LoginForm.style';

class LoginForm extends Component {
  static propTypes = {
    submitHandler: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
    classes: PropTypes.shape({
      textField: PropTypes.string.isRequired,
      button: PropTypes.string.isRequired
    }).isRequired,
    actions: PropTypes.shape({
      logUserIn: PropTypes.func
    }).isRequired
  };

  static defaultProps = {
    submitHandler: false
  };

  state = {
    username: null,
    password: null,
    apiLoading: false,
    apiMessage: null
  };

  /**
   * Handles field update action.
   *
   * @param {object} event - Field update event.
   */
  handleFieldUpdate = event => {
    const state = {};
    state[event.target.id] = event.target.value;
    this.setState(state);
  };

  /**
   * Handles a login form submit action.
   *
   * @param {object} event - Submit event object.
   */
  handleSubmit = event => {
    event.preventDefault();
    this.setState({ apiLoading: true });
    const { username, password } = this.state;
    const {
      actions: { logUserIn }
    } = this.props;

    logUserIn(username, password).catch(error => {
      this.setState({ apiLoading: false, apiMessage: error.toString() });
    });
  };

  /**
   * {@inheretdoc}
   */
  render() {
    const { classes, submitHandler } = this.props;
    const { apiLoading, apiMessage } = this.state;

    return (
      <form onSubmit={submitHandler || this.handleSubmit}>
        {apiMessage && <FormMessage>{apiMessage}</FormMessage>}
        <TextField
          id="username"
          label="Username"
          type="text"
          required
          onChange={this.handleFieldUpdate}
          className={classes.textField}
        />
        <TextField
          id="password"
          label="Password"
          type="password"
          required
          onChange={this.handleFieldUpdate}
          className={classes.textField}
        />
        <Button
          variant="raised"
          color="primary"
          type="submit"
          className={classes.button}
        >
          Log In
        </Button>
        {apiLoading && <FormLoading />}
      </form>
    );
  }
}

export default withStyles(LoginFormStyles)(LoginForm);
