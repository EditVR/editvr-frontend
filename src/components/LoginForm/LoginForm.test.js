/**
 * @file LoginForm.test.js
 * Contains tests for LoginForm.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import LoginForm from './LoginForm';

describe('<LoginForm />', () => {
  it('Matches its snapshot', () => {
    expect(
      renderer
        .create(
          <LoginForm />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
