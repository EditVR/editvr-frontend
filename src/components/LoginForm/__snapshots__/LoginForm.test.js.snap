// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Matches its snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-3 LoginForm-textField-1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
      data-shrink={false}
      htmlFor="username"
    >
      Username
      <span
        className="MuiFormLabel-asterisk-16"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-17 MuiInput-formControl-18 MuiInput-underline-21"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-25"
        defaultValue={undefined}
        disabled={false}
        id="username"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-3 LoginForm-textField-1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-12 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
      data-shrink={false}
      htmlFor="password"
    >
      Password
      <span
        className="MuiFormLabel-asterisk-16"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-17 MuiInput-formControl-18 MuiInput-underline-21"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-25 MuiInput-inputType-28"
        defaultValue={undefined}
        disabled={false}
        id="password"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value={undefined}
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-46 MuiButton-root-30 MuiButton-raised-36 MuiButton-raisedPrimary-37 LoginForm-button-2"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-31"
    >
      Login
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
  <a
    className="MuiButtonBase-root-46 MuiButton-root-30 MuiButton-raised-36 MuiButton-raisedSecondary-38 LoginForm-button-2"
    href="/login/reset"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-31"
    >
      Forgot Password
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </a>
</form>
`;
